client/scenes/stream_tracking.cpp

```diff
@@ -220,6 +239,32 @@ void scenes::stream::tracking()
 						t.resume();
 					}
 
+#ifdef __ANDROID__
+					if (next_battery_check < now)
+					{
+						from_headset::battery battery{};
+						auto intent = ctx.call<jni::object<"android/content/Intent">>("registerReceiver", receiver_jobj, filter_jobj);
+						if (intent)
+						{
+							auto level_jint = intent.call<jni::Int>("getIntExtra", level_jstr, default_jint);
+							auto scale_jint = intent.call<jni::Int>("getIntExtra", scale_jstr, default_jint);
+
+							if (level_jint && level_jint.value >= 0 && scale_jint && scale_jint.value >= 0)
+							{
+								battery.present = true;
+								battery.charge = (float)(level_jint.value) / (float)(scale_jint.value);
+							}
+
+							auto plugged_jint = intent.call<jni::Int>("getIntExtra", plugged_jstr, default_jint);
+							battery.charging = plugged_jint && plugged_jint.value > 0;
+						}
+
+						network_session->send_stream(battery);
+
+						next_battery_check = now + battery_check_interval;
+					}
```

Comment:
Is it possible to time/profile this block? It is unlikely to be an issue but the home-grown jni wrapper is likely slow.

We may also want to move that after the `busy_time` calculation or we may reduce polling rate if it takes more than 0.2ms.

Reply:
I'm getting an average of 0.8ms for the whole block. moved it after the busy time. let me see what else i can try.

Reply:
My guess is that `GetMethodID` inside `call` may be slow, a profiler would give more accurate data.

0.8ms every 5s is not bad but not entirely free. We should check if something can be done before merging.

Reply:
down to 0.5ms after caching the method id. i'm guessing it's the intent itself that takes long. should we look at moving this to a separate thread, or is it fine like this?

Reply:
I think it's ok like this. I'll try to profile it more accurately when I have time